server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionTaskManager.java

```diff
@@ -224,26 +207,32 @@ public synchronized boolean addTaskToWaitingQueue(AbstractCompactionTask compact
    * This method will submit task cached in queue with most priority to execution thread pool if
    * there is available thread.
    */
-  public synchronized void submitTaskFromTaskQueue() {
-    while (currentTaskNum.get()
-            < IoTDBDescriptor.getInstance().getConfig().getConcurrentCompactionThread()
-        && candidateCompactionTaskQueue.size() > 0) {
-      AbstractCompactionTask task = candidateCompactionTaskQueue.poll();
+  public void submitTaskFromTaskQueue() {
```

Comment:
It seems that this method is only invoked in Tests

Reply:
A thread is used in the Start function of CompactionTaskManager to execute this function